### PR TITLE
Feature/tools Bot and Slasher

### PR DESCRIPTION
**Pull request by invocamanman**

### CODE REVIEW

File: rollup-cli/src/wallet.js [hunk 1]
```diff
@@ -25,9 +25,9 @@ class Wallet {
    * To create a rollup wallet from mnemonic
    * @param {String} mnemonic - mnemonic
    */
-    static async fromMnemonic(mnemonic) {
-        const ethWallet = EthereumWallet.fromMnemonic(mnemonic);
-        const babyjubWallet = BabyJubWallet.fromMnemonic(mnemonic);
+    static async fromMnemonic(mnemonic, index= 0) {
```
Comment (krlosMata) [inline rollup-cli/src/wallet.js]: add comment on function `signRollupTx`

File: rollup-cli/tools/bot.js [hunk 2]
```diff
@@ -1,58 +1,67 @@
 const { Wallet } = require("../src/wallet");
-const { argv } = require('yargs'); // eslint-disable-line
+const { argv } = require("yargs").alias("t", "tokens").alias("mn", "mnemonic").alias("p", "path").alias("e", "ether")
+    .alias("w", "wallets").alias("d", "deposits").alias("s", "sends");
 const fs = require("fs");
-const { depositTx, sendTx } = require("../src/cli-utils");
-const configjson = "../src/resources/config.json";
+const configjsonBot = "../tools/resourcesBot/configBot.json";
+const ethers = require("ethers");
+const {createWallets} = require("../tools/createWallets");
```
Comment (krlosMata) [inline rollup-cli/tools/bot.js]: blank spaces before/after brackets:
`{ createWallet }`
Applies to this line and further lines

File: rollup-cli/tools/bot.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# Rollup-cli Bot
```
Comment (krlosMata) [inline rollup-cli/tools/bot.md]: - Review typo errors in this file:
  - `Automatic`
  - `necessary`
  - ...

File: rollup-cli/tools/bot.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+### Commands:
+- createwallets: Create, fund and save the wallets in files with encripted JSON format. 
+- deposit: Every wallet does a deposit Tx
+- send: Every wallet does a send Tx. Some Tx will be unvalid randomly
+- doall: Make the three options above
+
+
+### Previous configuration:
+It's necessari a configBot.json like this one:
```
Comment (krlosMata) [inline rollup-cli/tools/bot.md]: mark files names such as: `configBot.json`

File: rollup-cli/tools/resourcesBot/bad_transactoins.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+ # Possible bad send transactions
+
+ - Sign tx don't match with babyjub publick key
+ - Leaf don't have enough funds
+ - Idfrom / Idto don't exist in current tree
+ - Nonce d'ont match with the leaf nonce
```
Comment (krlosMata) [inline rollup-cli/tools/resourcesBot/bad_transactoins.md]: typo error:
`d'ont` to `do not`

File: rollup-cli/tools/resourcesBot/rollupabi.json [hunk 6]
```diff
@@ -0,0 +1,458 @@
+[
```
Comment (krlosMata) [inline rollup-cli/tools/resourcesBot/rollupabi.json]: resources such as `.abi` should be in `.gitignore`

File: rollup-cli/tools/resourcesBot/sendBot.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
+const axios = require("axios");
```
Comment (krlosMata) [inline rollup-cli/tools/resourcesBot/sendBot.js]: this file should not be in resources
Should be in `tools` folder

File: rollup-cli/tools/resourcesBot/walletFunder.json [hunk 8]
```diff
@@ -0,0 +1,47 @@
+{
```
Comment (krlosMata) [inline rollup-cli/tools/resourcesBot/walletFunder.json]: this file should be in `.gitignore`
We could add some examples files but it should be pointed out on file name: `walletFunder.example.json`
:smile:

File: rollup-cli/tools/slasher.test.js [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        abi = insRollupPoS.abi;
+        walletJson = JSON.parse(fs.readFileSync(walletPath, "utf8"));
+
+        db = new SMTMemDB();
+        rollupDB = await RollupDB(db);
+        
+    });
+    it("slash", async () => {
+        for(let i = 0; i <1011; i++) {
+            await forgeBlock();
```
Comment (krlosMata) [inline rollup-cli/tools/slasher.test.js]: we could use 
https://github.com/iden3/rollup/blob/master/test/contracts/helpers/timeTravel.js
to add blocks xD

File: rollup-cli/tools/walletSlash.json [hunk 10]
```diff
@@ -0,0 +1,47 @@
+{
```
Comment (krlosMata) [inline rollup-cli/tools/walletSlash.json]: This file should be in `.gitignore`

File: rollup-cli/tools/walletsDeposit.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        i++;
+    }
+    for(let i = 1; i <= files.length; i++){
+        for(let j = 0; j < numTransOnchain; j++) {
+            await deposit(node, addressRollup, Math.floor(amountToken/numTransOnchain), tokenId, await wallets[i].toEncryptedJson(passString), 
+                passString, JSON.parse(fs.readFileSync(abiRollupPath, "utf8")));
+            
+        }
+    }
+
```
Comment (krlosMata) [inline rollup-cli/tools/walletsDeposit.js]: do not leave blank lines :smile_cat: 
Applies to this file and the rest of the files

File: rollup-cli/README.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+Set a configuration file as the following: 
+#### Config.json (example)
+```
+{
+  "wallet": "./src/resources/wallet.json",
+  "operator": "http://127.0.0.1:9000",
+  "address": "0x3E03394a9d383312091a9c30BCe1354Eef737664",
+  "nodeEth": "http://localhost:8545",
+  "abi": "./src/resources/rollupabi.js"
```
[the remark this answers is not in the document]
Comment (invocamanman) [reply rollup-cli/README.md]: done

File: rollup-cli/tools/resourcesBot/sendBot.js (hunk 7 above)
Comment (invocamanman) [reply rollup-cli/tools/resourcesBot/sendBot.js]: Is in helpers now

**Pull request merged**